src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
First, they are not really independent issues - it would not make sense to specify SrcHasUpperZeros if you want to preserve the upper bits of the target. Second, if the caller is explicitly requesting VerbatimCopy it would be more logical to use:
```
inst_RV_RV(ins_Copy(targetType), targetReg, op1Reg, targetType, emitActualTypeSize(targetType));
```
instead of calling genSIMDScalarMove(), since at that point it's not a scalar move. So my preference would be to either keep the signature as-is, and have callers use the explicit copy if they want to copy the whole register, OR to add the 3rd state to the enum.